__tests__/app.test.js

```diff
@@ -9,77 +9,150 @@ beforeEach(() => {
-            const { users } = body
-            expect(Array.isArray(users)).toBe(true)
-            expect(users.length).toBe(4)
-            users.forEach((users) => {
-                expect(users).toHaveProperty('username', expect.any(String))
-                expect(users).toHaveProperty('name', expect.any(String))
-                expect(users).toHaveProperty('avatar_url', expect.any(String))
+            .get('/api/users')
+            .expect(200)
+            .then(({
+                body
+            }) => {
+                const {
+                    users
+                } = body
+                expect(Array.isArray(users)).toBe(true)
+                expect(users.length).toBe(4)
+                users.forEach((users) => {
+                    expect(users).toHaveProperty('username', expect.any(String))
+                    expect(users).toHaveProperty('name', expect.any(String))
+                    expect(users).toHaveProperty('avatar_url', expect.any(String))
+                })
             })
-        })
     })
 })
+
+describe('PATCH /api/articles', () => {
+    it('should return the object with an updated vote count by the specified amount', () => {
+        const updatedVote = {
+            inc_votes: 50
+        }
+        return request(app)
+            .patch(`/api/articles/4`)
+            .send(updatedVote)
+            .expect(200)
+            .then(({
+                body
+            }) => {
+                expect(body.article).toEqual({
+                    article_id: 4,
+                    title: "Student SUES Mitch!",
+                    topic: "mitch",
+                    author: "rogersop",
+                    body: "We all love Mitch and his wonderful, unique typing style. However, the volume of his typing has ALLEGEDLY burst another students eardrums, and they are now suing for damages",
+                    created_at: expect.any(String),
+                    votes: 50,
+                })
+            })
+    })
+    it('should return a 400 bad request if the votes are invalid such as not a number', () => {
+        const updatedVote = {
+            inc_votes: 'five'
+        }
+        return request(app)
+            .patch(`/api/articles/4`)
+            .send(updatedVote)
+            .expect(400)
+            .then(({
+                body
+            }) => {
+                expect(body).toEqual({
+                    msg: 'bad request'
+                })
+            })
+    })
+    it('should return a 404 route not found when given a valid id that does not exist', () => {
+        const updatedVote = {
+            inc_votes: 50
+        }
+        return request(app)
+            .patch(`/api/articles/567`)
+            .send(updatedVote)
+            .expect(404)
+            .then(({
+                body
+            }) => {
+                expect(body).toEqual({
+                    msg: 'article ID not found'
+                })
+            })
+    })
+})
```

Comment:
Some missing tests to add:

- [ ] 400 invalid id
- [ ]  200/400 inc_vote key missing - either respond with an error or make no change.